core/src/test/java/org/apache/iceberg/TestPartitioning.java

```diff
@@ -43,6 +43,12 @@ public class TestPartitioning {
           required(1, "id", Types.IntegerType.get()),
           required(2, "data", Types.StringType.get()),
           required(3, "category", Types.StringType.get()));
+  private static final PartitionSpec BY_DATA_SPEC =
+      PartitionSpec.builderFor(SCHEMA).identity("data").build();
+  private static final PartitionSpec BY_CATEGORY_DATA_SPEC =
+      PartitionSpec.builderFor(SCHEMA).identity("category").identity("data").build();
+  private static final PartitionSpec BY_DATA_CATEGORY_BUCKET_SPEC =
+      PartitionSpec.builderFor(SCHEMA).identity("data").bucket("category", 8).build();
 
```

Comment:
Tests all look good to me, I think you covered all the cases we were discussing yesterday. One nit is that we could probably parameterize this test suite on format version, it looks like we expect the same output and run the same code so we could shorten this file a bit with the parameterization.

Reply:
There are some slight differences in how other tests behave for different format versions. We can cover that with assumes and v1 and v2 checks. Since it is an existing class, though, I'd consider refactoring in a separate PR.